### ISSUE TITLE
Minimum # of pixels for training

### ISSUE THREAD

**Issue by choosehappy** [choosehappy/QuickAnnotator#128]
I think that this line:

https://github.com/choosehappy/QuickAnnotator/blob/e3dd5834081af379d718bec6820936495c6e991f/quickannotator/dl/dataset.py#L154

Will essentially not produce _any_ patches ever if there are no annotations at all -- though now we have some losses (e.g., reconstruction, embedding, etc) that could still make some "progress" - but perhaps not meaningful progress.

Should we reduce this constraint?

relatedly, this is where the tiles come from:

https://github.com/choosehappy/QuickAnnotator/blob/e3dd5834081af379d718bec6820936495c6e991f/quickannotator/db/crud/tile.py#L312

but i can't tell from this snippet what/when tiles for this annotation class are put in the table?

i.e., if the user doesn't annotate anything, are there still tiles in this DB for querying, or does this query always return None until there is at least a single annotation -- in which case the line of code above for patch filtering will work as expected?

i'm wondering what the "base case" here is when nothing is annotated - is the system doing some DL training - or is it all just waiting for the user to provide at least 1 annotation?

@jacksonjacobs1  thoughts?

**Comment by jacksonjacobs1** [comment]
In short, the query will return None until there is at least a single annotation. The only caveat is that we don't currently "reset" the tile, e.g., set gt_datetime to None when all of the annotations have been removed. So you can end up with an empty tile, in which case the first line of code you linked to will kick in. Does that make sense?

**Comment by choosehappy** [comment]
thanks, yes, makes sense. the system should now be able to handle empty tiles and still be able to make process on fine-tunning the model - akin more to pretraining. so there is some value into sending the model something, even if its empty --- however, i guess our question is -how do we know what / where to send stuff? this seems like a separate orchestration problem that would require a lot of work to address - we basically wouldw ant to say "if there are no annotations, then grab tiles randomly from within the tissue mask area, however, as soon as there are annotations, stop doing that and focus on just those tiles". easy to write, but i feel like would require a disproportionate amount of effort to address.  at the same time - while it would for sure improve the model - how much time is the user really going to be spending looking at a WSI and not annotating a _single_ thing? as soon as there is at least 1 annotation, we exit that edge case - so its likely sufficiently rare that it may not be worth the effort to implement in a v1
